lib/LaTeXML/Common/Color.pm

```diff
@@ -119,14 +119,33 @@ sub complement {
 # Mix $self*$fraction + $color*(1-$fraction)
 sub mix {
   my ($self, $color, $fraction) = @_;
-  $color = $color->convert($self->model) unless $self->model eq $color->model;
+  # mixing 'rgb|cmy' with 'gray' should yield 'rgb|cmy'.
+  my $base_model = $self->model;
+  my $mix_model  = $color->model;
+  if ($base_model ne $mix_model) {
+    # careful, if "grayscale" remains the base, an incoming color-rich model will
+    # have components discarded.
+    if ($base_model eq 'gray') {
+      my $enriched_base = $self->convert($mix_model);
+      return $enriched_base->mix($color, $fraction);
```

Comment:
Interesting; looks like you've likely got the right idea. I'm wondering out-loud whether it might be slightly less convoluted than a return from the middle to start with `my $base = $self;` and then if gray, do `$base = $self->convert($mix_model);`, and do the last few lines with `$base` instead of `$self` ?  That would seem to make more transparent that you're really just re-expressing self in a different color model.

Reply:
Good point, the extra recursion is just needless performance tax here. Now adjusted.